pydis_core/utils/scheduling.py

```diff
@@ -243,15 +246,29 @@ def create_task(
 
     _background_tasks.add(task)
     task.add_done_callback(_background_tasks.discard)
-    task.add_done_callback(partial(_log_task_exception, suppressed_exceptions=suppressed_exceptions))
+    task.add_done_callback(
+        partial(
+            asyncio.run,
+            partial(_log_task_exception, task, suppressed_exceptions=suppressed_exceptions)
+        )
+    )
     return task
 
 
-def _log_task_exception(task: asyncio.Task, *, suppressed_exceptions: tuple[type[Exception], ...]) -> None:
-    """Retrieve and log the exception raised in ``task`` if one exists."""
+async def _log_task_exception(task: asyncio.Task, *, suppressed_exceptions: tuple[type[Exception], ...]) -> None:
+    """Retrieve and log the exception raised in ``task`` if one exists and it's not suppressed/handled."""
     with contextlib.suppress(asyncio.CancelledError):
         exception = task.exception()
-        # Log the exception if one exists.
+        # Log the exception if one exists and it's not suppressed/handled.
         if exception and not isinstance(exception, suppressed_exceptions):
+            if isinstance(exception, Forbidden):
+                try:
+                    await handle_forbidden_from_block(exception)
```

Comment:
Could we call asyncio.run on this line directly, rather than change the function to async and having that mash of partials when adding the call back?

Reply:
I'm confused how using asyncio.run would work here. Doesn't that try to create a new event loop, which is a problem since we already have a running loop on that thread?

Reply:
No, we cannot use `run_until_complete` because the loop is already running.

One solution may be to wrap the task being scheduled with another coroutine that awaits the original coroutine and the error handler afterwards. But that might be messy, particularly if we need to be careful about handling task cancellation.

Reply:
Addressed in 467867f.